src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_raster/LocalServerDynamicWorkspaceRasterSample.java

```diff
@@ -0,0 +1,164 @@
+import com.esri.arcgisruntime.localserver.RasterWorkspace;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+import javafx.stage.Stage;
+
+public class LocalServerDynamicWorkspaceRasterSample extends Application {
+
+  private MapView mapView;
+  private static final LocalServer server = LocalServer.INSTANCE;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Local Server Dynamic Workspace Raster Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a control panel
+      VBox vBoxControl = new VBox(6);
+      vBoxControl.setMaxSize(240, 185);
+      vBoxControl.getStyleClass().add("panel-region");
+
+      // create the descriptive label
+      Label description = new Label("This application shows how to create a dynamic workspace connection " +
+              "to a raster folder and display raster data in a map. Click the " +
+              "button to select a local raster and add it to the map.");
+      description.setWrapText(true);
+
+      // create a file chooser to select a raster
+      FileChooser fileChooser = new FileChooser();
+      fileChooser.setTitle("Open Resource File");
+      fileChooser.getExtensionFilters().addAll(
+          new ExtensionFilter("Image Files", "*.tif"));
+      fileChooser.setInitialDirectory(new File("./samples-data/raster/"));
+
+      // create button to open file chooser
+      Button addButton = new Button("Choose Raster");
+      addButton.setMaxWidth(Double.MAX_VALUE);
+      addButton.setDisable(true);
+      addButton.setOnAction(e -> {
+       File rasterFile = fileChooser.showOpenDialog(stage);
+       if(rasterFile != null){
+         startLocalService(rasterFile);
+       }
+      });
+
+      // add button to the control panel
+      vBoxControl.getChildren().addAll(description, addButton);
+
+      // create a map and add it to a map view
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // start the local server
+      server.startAsync().addDoneListener(() -> {
```

Comment:
On developer site best practices says we should first check that the local server is installed.
```Java
 // check that local server install path can be accessed
      if (LocalServer.INSTANCE.checkInstallValid()) {
        server = LocalServer.INSTANCE;
        // start the local server
        server.startAsync().addDoneListener(() -> {
          if (server.getStatus() == LocalServerStatus.STARTED) {
            addButton.setDisable(false);
          } else {
            // should display another message here like below for if it doesn't start properly
          }
        });
      } else {
        Platform.runLater(() -> {
          Alert dialog = new Alert(AlertType.INFORMATION);
          dialog.setHeaderText("Local Server Load Error");
          dialog.setContentText("Local Server install path couldn't be located.");
          dialog.showAndWait();

          Platform.exit();
        });
      }
```